packages/engine-ui/src/components/PlanBookingDetails.tsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react'
+import styled from 'styled-components'
+import { useHistory, useParams } from 'react-router-dom'
+import Elements from '../shared-elements'
+import helpers from '../utils/helpers'
+import { Booking } from '../types'
+import MainRouteLayout from './layout/MainRouteLayout'
+import ContactPhone from '../assets/contact-phone.svg'
+import ContactName from '../assets/contact-name.svg'
+import moment from 'moment'
+import { UIStateContext } from '../utils/UIStateContext'
+
+interface Props {
+  bookings: Booking[]
+}
+
+const Paragraph = styled.p`
+  margin-bottom: 0.25rem;
+  margin-top: 0;
+`
+
+const CapitalizeParagraph = styled(Paragraph)`
+  text-transform: capitalize;
+`
+
+const PlanBookingDetails = ({ bookings }: Props) => {
+  const { dispatch } = React.useContext(UIStateContext)
+  const history = useHistory()
+  const { routeId, bookingId } = useParams<{
+    routeId: string
+    bookingId: string
+  }>()
+
+  React.useEffect(
+    () => () => {
+      dispatch({ type: 'highlightBooking', payload: undefined })
+    },
+    [dispatch]
+  )
+
+  const [booking] = bookings.filter((d: any) => d.id === bookingId)
+
+  const handleDeleteClick = (bookingId: string) => {
+    if (
+      window.confirm(
+        'Är du säker på att du vill ta bort bokningen från den föreslagna planen?'
+      )
+    ) {
+      //function for removing a booking from a plan goes here.
+      return history.push(`/plans/routes/${routeId}`)
+    }
+  }
+
+  if (!booking || !bookings) return <p>Laddar bokning...</p>
+
+  return (
+    <MainRouteLayout redirect={`/plans/routes/${routeId}`}>
+      <Elements.Layout.Container>
+        <Elements.Layout.FlexRowWrapper>
+          <h3>Bokning</h3>
+          <Elements.Typography.RoundedLabelDisplay margin="0 0.5rem">
+            {helpers.withoutLastFourChars(bookingId)}
+            <Elements.Typography.SpanBold>
+              {helpers.getLastFourChars(bookingId)}
+            </Elements.Typography.SpanBold>
+          </Elements.Typography.RoundedLabelDisplay>
+        </Elements.Layout.FlexRowWrapper>
+        <Elements.Layout.SectionWithMargin>
+          {booking.metadata.cargo && (
+            <CapitalizeParagraph>
+              <Elements.Typography.SpanBold>
+                Innehåll:{' '}
+              </Elements.Typography.SpanBold>
+              {booking.metadata.cargo}
+            </CapitalizeParagraph>
+          )}
+          <Paragraph>
+            <Elements.Typography.SpanBold>
+              Ömtåligt:{' '}
+            </Elements.Typography.SpanBold>
+            {booking.metadata.fragile ? 'Ja' : 'Nej'}
+          </Paragraph>
+          {booking.size.measurement && (
```

Comment:
In types aint nullable

Reply:
We should ensure that everything in types.ts is correct, for the nullable fields, so that we are making correct checks. This goes for many of the fields